Website/src/rendering/src/components/PreviewSearch/PreviewSearch.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import ClickOutside from './ClickOutside';
+import LeftColumn from './LeftColumn';
+import RightColumn from './RightColumn';
+
+function debounce(
+  func: (arg: string | { value: string }) => void,
+  wait: number,
+  immediate: boolean
+) {
+  let timeout: NodeJS.Timeout;
+
+  return function returnFn(this: unknown, ...rest: unknown[]) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const context = this;
```

Comment:
To avoid the need for the `context` variable, the `later` function can be defined with a fat arrow instead of the `function` keyword.

```javascript
const later = () => {
  // can use this inside here. It will be the parent scope this object.
}
```

I am unsure about the second use of the `context` variable though.